broker.go

```diff
@@ -156,6 +159,39 @@ func NewBroker(addr string) *Broker {
 	return &Broker{id: -1, addr: addr}
 }
 
+func (b *Broker) getSockOpt(conn *net.TCPConn, opt int) (int, error) {
+	file, err := conn.File()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get file from connection: %w", err)
+	}
+	defer file.Close() // Close the duplicated file descriptor
+
+	sockErr, err := sockopt.GetSockoptInt(file.Fd(), opt)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get sockopt %d: %w", opt, err)
+	}
+
+	return sockErr, nil
+}
+
+func (b *Broker) getSockError() error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	if b.connTCP == nil {
+		return nil
+	}
+
+	sockErr, err := b.getSockOpt(b.connTCP, sockopt.SockoptError)
+	if err != nil {
+		return fmt.Errorf("failed to get socket error: %w", err)
+	}
+	if sockErr != 0 {
+		return syscall.Errno(sockErr)
+	}
+	return nil
+}
```

Comment:
As mentioned, using `SyscallConn` and `RawConn.Control` remove the need to `dup` the underlying file descriptor: 
```golang
func (b *Broker) getSockError() error {
	b.lock.Lock()
	defer b.lock.Unlock()

	if conn, _ := b.conn.(*net.TCPConn); conn != nil {
		rawConn, err := conn.SyscallConn()
		if err != nil {
			return err
		}

		var sockErr int
		var opErr error

		err = rawConn.Control(func(fd uintptr) {
				sockErr, opErr = sockopt.GetSockoptInt(fd, sockopt.SockoptError)
		})
		if err != nil {
			return fmt.Errorf("failed to get socket error: %w", err)
		}

		if opErr != nil {
			return fmt.Errorf("failed to get socket error: %w", err)
		}

		if sockErr != 0 {
			return syscall.Errno(sockErr)
		}
	}

	return nil
}
```

Reply:
LGTM.

I'll update it like this.

Thanks!